src/algebra/parity.lean

```diff
@@ -82,17 +82,54 @@ by { rintro ⟨c, rfl⟩ a, simp_rw [←two_mul, pow_mul, neg_sq] }
 
 lemma even.neg_one_pow (h : even n) : (-1 : α) ^ n = 1 := by rw [h.neg_pow, one_pow]
 
+@[simp]
+lemma is_square_zero [mul_zero_class α] : is_square (0 : α) := ⟨0, (mul_zero _).symm⟩
```

Comment:
Can you move that before the `section monoid`? Here your lemma is taking the nonsense `[monoid α] [mul_zero_class α]`.

Reply:
Done!